### ISSUE TITLE
Not Able install on custom VPC on GCP

### ISSUE THREAD

**Issue by anoopl** [openshift/installer#4006]
# Version
4.5.4
```console
$ ./openshift-install 4.5.4
built from commit 01f5643a02f154246fab0923f8828aa9ae3b76fb
release image quay.io/openshift-release-dev/ocp-release@sha256:02dfcae8f6a67e715380542654c952c981c59604b1ba7f569b13b9e5d0fbbed3```
 Same issue with version 4.5.3
# Platform:
GCP

* IPI 

# What happened?

We are trying to install latest Openshift with openshift-installer using IPI But we have a custom VPC and Subnets:
machine network: 10.152.0.0/16
master subnet: 10.152.0.0/19
worker subnet: 10.152.32.0/19
jump host:   10.152.64.0/19

Installer fails with below logs. Is it because we use machine network: 10.152.0.0/16. It can be only set as 10.0.0.0/16
We have same issue with installer 4.5.3 as well. We tried both Publish bith External and Internal. Everything is created if we let the installer create VPC and subnets with it's own range and machine network is 10.0.0.0/16.

We are able to reach the API server with curl and get the version. When I try oc get pods --all-namespaces all pods are pending. When I look at Master by SSH, I can see that Kubelet is not started. But I could not find much information from the Kubelet log.



 `INFO Waiting up to 40m0s for bootstrapping to complete...
E0804 10:38:34.013369    6948 reflector.go:307] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to watch *v1.ConfigMap: Get https://api.ocp45.thelettuce.co:6443/api/v1/namespaces/kube-system/configmaps?allowWatchBookmarks=true&fieldSelector=metadata.name%3Dbootstrap&resourceVersion=3762&timeoutSeconds=502&watch=true: dial tcp 10.152.0.22:6443: connect: connection refused
E0804 10:38:35.015302    6948 reflector.go:307] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to watch *v1.ConfigMap: Get https://api.ocp45.thelettuce.co:6443/api/v1/namespaces/kube-system/configmaps?allowWatchBookmarks=true&fieldSelector=metadata.name%3Dbootstrap&resourceVersion=3762&timeoutSeconds=574&watch=true: dial tcp 10.152.0.22:6443: connect: connection refused
E0804 10:38:36.018109    6948 reflector.go:307] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to watch *v1.ConfigMap: Get https://api.ocp45.thelettuce.co:6443/api/v1/namespaces/kube-system/configmaps?allowWatchBookmarks=true&fieldSelector=metadata.name%3Dbootstrap&resourceVersion=3762&timeoutSeconds=599&watch=true: dial tcp 10.152.0.22:6443: connect: connection refused`.

# What you expected to happen?

Install Openshift on a custom VPC 

# How to reproduce it (as minimally and precisely as possible)?

Create a VPC, create subnets with range: 

```console
machine network: 10.152.0.0/16
master subnet: 10.152.0.0/19
worker subnet: 10.152.32.0/19
jump host:   10.152.64.0/19
```

**Comment by anoopl** [comment]
Additional information:
Attached install log
[log-bundle-20200804105759.tar.gz](https://github.com/openshift/installer/files/5021698/log-bundle-20200804105759.tar.gz)

Logs from one of the master node:
---
[systemd]
Failed Units: 2
  machine-config-daemon-firstboot.service
  machine-config-daemon-host.service
[core@ocp45-fllws-master-0 ~]$ sudo journalctl -u kubelet
-- Logs begin at Tue 2020-08-04 10:15:30 UTC, end at Tue 2020-08-04 11:03:33 UTC. --
Aug 04 10:26:01 ocp45-fllws-master-0.europe-west3-a.c.ab-gcp-lab-openshift-custo systemd[1]: Dependency failed for MCO environment configuration.
Aug 04 10:26:01 ocp45-fllws-master-0.europe-west3-a.c.ab-gcp-lab-openshift-custo systemd[1]: kubelet.service: Job kubelet.service/start failed with result 'dependency'.

I guess cri-o not starting
[root@ocp45-fllws-master-0 system]# systemctl status crio
● crio.service - MCO environment configuration
   Loaded: loaded (/usr/lib/systemd/system/crio.service; disabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/crio.service.d
           └─10-mco-default-env.conf
   Active: inactive (dead)
     Docs: https://github.com/cri-o/cri-o

Aug 04 10:26:01 ocp45-fllws-master-0.europe-west3-a.c.ab-gcp-lab-openshift-custo systemd[1]: Dependency failed for MCO environment configuration.
Aug 04 10:26:01 ocp45-fllws-master-0.europe-west3-a.c.ab-gcp-lab-openshift-custo systemd[1]: crio.service: Job crio.service/start failed with result 'dependency'.

**Comment by abhinavdahiya** [comment]
You should set the networks for machine in the install config

```
networking:
  machineNetwork:
  - cidr:  10.152.0.0/16
```

**Comment by anoopl** [comment]
@abhinavdahiya 
Thanks for the reply. I had already specified that. Now I tried another time by changing subnets and 
networking:
  machineNetwork:
  - cidr:  10.0.0.0/16

To see if it works with default network. That did not work as well.
So I am able to create cluster if I do not provide any VPC network and let installer create the VPC and subnets.
It's not working if I provide a VPC and subnet names

**Comment by abhinavdahiya** [comment]
> @abhinavdahiya
> Thanks for the reply. I had already specified that. Now I tried another time by changing subnets and
> networking:
> machineNetwork:
> 
>     * cidr:  10.0.0.0/16
> 
> 
> To see if it works with default network. That did not work as well.
> So I am able to create cluster if I do not provide any VPC network and let installer create the VPC and subnets.
> It's not working if I provide a VPC and subnet names

Just to clarify, 

Previously you were using 
```
master subnet: 10.152.0.0/19
worker subnet: 10.152.32.0/19
```

and you set the appropriate fields in install-config
```
networking:
  machineNetwork:
  - cidr:  10.152.0.0/16
platform:
  network:
  controlPlaneSubnet: <master subnet>
  computeSubnet: <worker subnet>
```

And it failed to bootstrap the cluster and https://github.com/openshift/installer/files/5021698/log-bundle-20200804105759.tar.gz is the debugging log bundle from that, correct?

**Comment by anoopl** [comment]
@abhinavdahiya , Yes I had specified all these values, yes, this log is correct. I found one issue when comparing with the one working with no VPC specified. There are no NAT Gateways on the one working created by the installer. Which is missing on the one with custom VPC. So do I have to create this manually? Let me try again after creating these two NAT Gateways manually.

**Comment by abhinavdahiya** [comment]
So looking at the log-bundle,


control-plane/<ip>/journals/machine-config-daemon-host.service
```
Aug 04 10:17:26 ocp45-fllws-master-1.europe-west3-b.c.ab-gcp-lab-openshift-custo podman[1878]: 2020-08-04 10:17:26.523305216 +0000 UTC m=+0.480429533 system refresh
Aug 04 10:18:26 ocp45-fllws-master-1.europe-west3-b.c.ab-gcp-lab-openshift-custo machine-config-daemon[1865]: Error: error pulling image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:77e9ace116cec652637a79449dea00c6b4af5463ca2474463549cf145bc44438": unable to pull quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:77e9ace116cec652637a79449dea00c6b4af5463ca2474463549cf145bc44438: unable to pull image: Error initializing source docker://quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:77e9ace116cec652637a79449dea00c6b4af5463ca2474463549cf145bc44438: error pinging docker registry quay.io: Get https://quay.io/v2/: dial tcp 54.236.134.150:443: i/o timeout
```

So the bootstrap host has the public IP and therefore can access the internet, but the control-plane hosts do not have public IP and therefore need a explicit way to access the internet.
see the docs for https://docs.openshift.com/container-platform/4.5/installing/installing_gcp/installing-gcp-vpc.html#cluster-entitlements_installing-gcp-vpc `You must have internet access to:` section.

see https://cloud.google.com/solutions/building-internet-connectivity-for-private-vms for how to setup that.
the installer uses a Cloud NAT when it creates VPC for the user, and you can also setup something like that up for your existing VPCs. https://cloud.google.com/solutions/building-internet-connectivity-for-private-vms#deploying_cloud_nat_for_fetching

**Comment by anoopl** [comment]
@abhinavdahiya We create a NAT Gateway for both master and worker subnets. Now Installation is progressing. But a new error:

```EBUG Still waiting for the cluster to initialize: Working towards 4.5.4: 76% complete
DEBUG Still waiting for the cluster to initialize: Multiple errors are preventing progress:
* Cluster operator console is reporting a failure: DefaultIngressCertValidationDegraded: default-ingress-cert configmap not found
* Could not update prometheusrule "openshift-cloud-credential-operator/cloud-credential-operator-alerts" (186 of 584): the server does not recognize this resource, check extension API servers
* Could not update prometheusrule "openshift-cluster-samples-operator/samples-operator-alerts" (303 of 584): the server does not recognize this resource, check extension API servers
* Could not update servicemonitor "openshift-authentication-operator/authentication-operator" (499 of 584): the server does not recognize this resource, check extension API servers
* Could not update servicemonitor "openshift-cluster-machine-approver/cluster-machine-approver" (509 of 584): the server does not recognize this resource, check extension API servers
* Could not update servicemonitor "openshift-cluster-version/cluster-version-operator" (8 of 584): the server does not recognize this resource, check extension API servers
* Could not update servicemonitor "openshift-config-operator/config-operator" (113 of 584): the server does not recognize this resource, check extension API servers
* Could not update servicemonitor "openshift-image-registry/image-registry" (505 of 584): the server does not recognize this resource, check extension API servers
* Could not update servicemonitor "openshift-insights/insights-operator" (409 of 584): the server does not recognize this resource, check extension API servers
* Could not update servicemonitor "openshift-machine-api/cluster-autoscaler-operator" (218 of 584): the server does not recognize this resource, check extension API servers
* Could not update servicemonitor "openshift-marketplace/marketplace-operator" (444 of 584): the server does not recognize this resource, check extension API servers
```
Any idea?

**Comment by anoopl** [comment]
Even with this error, we had completed the setup and everything looks okay if we can ignore those errors 👍

**Comment by abhinavdahiya** [comment]
Yes these seem like an intermittent state and as long as the install complete with no error, we can ignore these that happen in the meanwhile..

**Comment by abhinavdahiya** [comment]
> Even with this error, we had completed the setup and everything looks okay if we can ignore those errors +1

> Yes these seem like an intermittent state and as long as the install complete with no error, we can ignore these that happen in the meanwhile..

/close
